src/main/java/seedu/address/logic/parser/editeventcommand/EventDescriptor.java

```diff
@@ -0,0 +1,105 @@
+package seedu.address.logic.parser.editeventcommand;
+
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.parser.editpersoncommandsparser.PersonDescriptor;
+import seedu.address.model.event.fields.DateTime;
+import seedu.address.model.event.fields.Description;
+import seedu.address.model.event.fields.Recurrence;
+
+/**
+ * Stores the details of the event.
+ */
+public class EventDescriptor {
+
+    private Optional<Index> index;
```

Comment:
```suggestion
    private Index index;
```

I think we dont need an Optional<Index> here. I used Optional<Index> for the PersonDescriptor because the EditUserCommand may not have used Index. But since EditEventCommand always uses the Index, no need to add another layer of complexity to it. 

Remember to change the rest of your methods to fit the change